### PR TITLE
drivers: npmx: npmx_shell: add buck mode setter command

### PR DESCRIPTION
**Pull request by wilu-nordic**
JIRA: NRFX-3850

DNM until:
* https://github.com/NordicSemiconductor/npmx-zephyr/pull/18

### CODE REVIEW

**Review by masz-nordic — APPROVED** [review]

**Comment by kylebonnici** [comment]
Does this PR add PFM and PWM to `npmx buck gpio pwm_force set ....`
My understanding this that we needed to add PFM and PWN to the existing GPIO[n] list

File: drivers/npmx/npmx_shell.c [hunk 1]
```diff
@@ -936,6 +936,68 @@ static int cmd_buck_gpio_forced_pwm_set(const struct shell *shell, size_t argc,
...
+		return 0;
+	}
+
+	if (argc < 3) {
+		shell_error(shell, "Error: missing buck mode.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t buck_indx = (uint8_t)shell_strtoul(argv[1], 0, &err);
```
Comment (mkleszcz-no) [inline drivers/npmx/npmx_shell.c:959]: 256 will be a valid value here, please take a look at this -> https://github.com/nrfconnect/npmx/pull/112

Comment (mkleszcz-no) [reply drivers/npmx/npmx_shell.c:959]: Or just let me know when it will be merged - I will align it within mine PR

**Review by mkleszcz-no — APPROVED** [review]

**Comment by wilu-nordic** [comment]
There is no getter function due to the fact that PWM mode is controlled by `task` register (can not be red).
image

Only reading PFM or auto is available:
image

So it won't be obvious when the user sets PWM but never get PWM.

**Pull request merged**